src/lib/operators/maintenance/create_index.cpp

```diff
@@ -0,0 +1,94 @@
+#include "create_index.hpp"
+
+#include <sstream>
+
+#include "constant_mappings.hpp"
+#include "hyrise.hpp"
+#include "operators/insert.hpp"
+#include "storage/table.hpp"
+#include "storage/index/group_key/group_key_index.hpp"
+#include "utils/assert.hpp"
+
+
+namespace opossum {
+
+CreateIndex::CreateIndex(const std::string& init_index_name,
+                         const std::shared_ptr<const std::vector<ColumnID>>& init_column_ids,
+                         const bool init_if_not_exists,
+                         const std::shared_ptr<const AbstractOperator>& input_operator)
+    : AbstractReadWriteOperator(OperatorType::CreateIndex, input_operator),
+      index_name(init_index_name),
+      column_ids(init_column_ids),
+      if_not_exists(init_if_not_exists)
+    {}
+
+const std::string& CreateIndex::name() const {
+  static const auto name = std::string{"CreateIndex"};
+  return name;
+}
+
+std::string CreateIndex::description(DescriptionMode description_mode) const {
+  // TODO: craft usefull description print output
+  /*
+  std::ostringstream stream;
+
+  const auto* const separator = description_mode == DescriptionMode::SingleLine ? ", " : "\n";
+
+  // If the input operator has already been cleared, we cannot retrieve its columns anymore. However, since the table
+  // has been created, we can simply pull the definitions from the new table.
+  const auto column_definitions = left_input_table()
+                                  ? left_input_table()->column_definitions()
+                                  : Hyrise::get().storage_manager.get_table(table_name)->column_definitions();
+
+  stream << AbstractOperator::description(description_mode) << " '" << table_name << "' (";
+  for (auto column_id = ColumnID{0}; column_id < column_definitions.size(); ++column_id) {
+    const auto& column_definition = column_definitions[column_id];
+
+    stream << "'" << column_definition.name << "' " << column_definition.data_type << " ";
+    if (column_definition.nullable) {
+      stream << "NULL";
+    } else {
+      stream << "NOT NULL";
+    }
+
+    if (column_id + 1u < column_definitions.size()) {
+      stream << separator;
+    }
+  }
+  stream << ")";
+
+  return stream.str();
+   */
+  return std::string("create index description");
+}
+
+std::shared_ptr<const Table> CreateIndex::_on_execute(std::shared_ptr<TransactionContext> context) {
+  auto table_to_be_indexed = std::const_pointer_cast<Table>(left_input()->get_output());
```

Comment:
If you check the description of the `GetTable` operator, you will find the following information:
>The returned table is NOT
// the same table as stored in the StorageManager. If that stored table is changed (most importantly: if a chunk is
// added), this is not reflected in GetTable's result.

In addition, changes that you apply on the output table of the `GetTable` operator are not applied on the base table that is stored in the `StorageManager`. In other words: we need to get the table from the storage manager during the execution of the `CreateIndex` operator. Therefore, we actually don't need the output of the `GetTable` operator. We thus can omit the `GetTable` operator and pass the required information, i.e., the table name, directly to the `CreateTable` operator.

Reply:
Thank you for the hint, we refactored it, and it works now! 🔥